### PR TITLE
YTS-3 creating home page

### PR DESCRIPTION
**Pull request by Tayyab-athar-1998**

### CODE REVIEW

**Comment by Tayyab-athar-1998** [comment]
TODO: Move styles from application.scss to home.scss(classes in home.scss were not applying on the view)

**Comment by Tayyab-athar-1998** [comment]
> TODO: Move styles from application.scss to home.scss(classes in home.scss were not applying on the view)

Done!

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class HomeController < ApplicationController
+  def index
+    @movies_on_number_of_downlaods = Movie.order_by_number_of_downloads.take(4)
```
Comment (IDaniyal) [inline app/controllers/home_controller.rb]:
```suggestion
    @popular_movies = Movie.order_by_number_of_downloads.take(4)
```

Comment (usmansaleem001) [inline app/controllers/home_controller.rb]:
```suggestion
    @movies_on_number_of_downlaods = Movie.most_downloaded.take(4)
```

Comment (usmansaleem001) [reply app/controllers/home_controller.rb]: move 4 to constant and use it.

File: app/models/movie.rb [hunk 2]
```diff
@@ -18,6 +18,9 @@ class Movie < ApplicationRecord
     message: '%<value> is not a allowed language'
   }
 
+  scope :order_by_number_of_downloads, -> { order(number_of_downloads: :desc) }
+  scope :latest_movies, -> { order(release_date: :desc) }
+
   serialize :video_quality
```
Comment (usmansaleem001) [inline app/models/movie.rb]: serialize should be on top.

File: app/views/home/index.html.haml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#homepage-container.container-fluid
+  .popular-downloads-container
+    .container
+      .row
+        %h2.popular-downloads-wrapper
+          Popular Downloads
+      =render partial: "shared/movie_row", locals: {movies: @movies_on_number_of_downlaods}
```
Comment (usmansaleem001) [inline app/views/home/index.html.haml]: prefer single quotes

File: app/views/shared/_header.html.haml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+-# TODO: styling header by looking into yts.mx
+%nav.navbar.navbar-expand-sm.bg-dark.navbar-dark
+  = image_tag  "logo.svg"
+  %ul.navbar-nav.ml-auto
+    %li.nav-item
+      %form.form-inline
+        %input.form-control.rounded-5{:placeholder => 'Quick search'}
+    %li.nav-item
+      %a.nav-link{:herf => '#'} Home
```
Comment (usmansaleem001) [inline app/views/shared/_header.html.haml]: new hash notation

File: app/views/shared/_header.html.haml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+-# TODO: styling header by looking into yts.mx
+%nav.navbar.navbar-expand-sm.bg-dark.navbar-dark
+  = image_tag  "logo.svg"
+  %ul.navbar-nav.ml-auto
+    %li.nav-item
+      %form.form-inline
+        %input.form-control.rounded-5{:placeholder => 'Quick search'}
```
Comment (usmansaleem001) [inline app/views/shared/_header.html.haml]: new notation

File: app/views/shared/_movie_brick.html.haml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+- content_for :head do
+  %link{:rel => 'stylesheet',:href => '/assets/movie_brick.css'}
+
+%div
+  %div
+    = image_tag  "default.png"
+  %div
+    %p{style: 'color: #fff'}
+      = movie.name
+    %p{style: 'color: #919191'}
```
Comment (usmansaleem001) [inline app/views/shared/_movie_brick.html.haml:10]: the style shouldn't be defined here.

**Review by usmansaleem001 — CHANGES_REQUESTED** [review]

**Pull request merged**